src/drivers/imu/adis16497/ADIS16497.cpp

```diff
@@ -247,7 +249,19 @@ ADIS16497::probe()
 			PX4_DEBUG("PRODUCT: %X", product_id);
 
 			if (self_test()) {
-				return PX4_OK;
+
+				// Switch to config page
+				write_reg16(PAGE_ID, 0x03);
+
+				uint16_t model_id = read_reg16(RANG_MDL);
```

Comment:
You might want to clear everything but the lower 4 bits. The datasheet just says the reset aren't used.